pkg/background/generate/generate.go

```diff
@@ -610,6 +643,121 @@ func manageClone(log logr.Logger, apiVersion, kind, namespace, name, policy stri
 	return obj.UnstructuredContent(), Create, nil
 }
 
+func manageCloneList(log logr.Logger, namespace, policy string, clone kyvernov1.Generation, client dclient.Interface) []GenerateResponse {
+	var response []GenerateResponse
+
+	rNamespace := clone.CloneList.Namespace
+	if rNamespace == "" {
+		// return nil, Skip, fmt.Errorf("failed to find source namespace")
+		response = append(response, GenerateResponse{
+			Data:   nil,
+			Action: Skip,
+			Error:  fmt.Errorf("failed to find source namespace"),
+		})
+	}
+
+	kinds := clone.CloneList.Kinds
+	if len(kinds) == 0 {
+		//		return nil, Skip, fmt.Errorf("failed to find kinds list")
+		response = append(response, GenerateResponse{
+			Data:   nil,
+			Action: Skip,
+			Error:  fmt.Errorf("failed to find kinds list"),
+		})
+	}
+
+	for _, kind := range kinds {
+		apiVersion, kind := kubeutils.GetKindFromGVK(kind)
+		resources, err := client.ListResource(apiVersion, kind, rNamespace, nil)
+		if err != nil {
+			//return nil, Skip, fmt.Errorf("failed to list resource %s %s/%s. %v", apiVersion, kind, rNamespace, err)
+			response = append(response, GenerateResponse{
+				Data:   nil,
+				Action: Skip,
+				Error:  fmt.Errorf("failed to list resource %s %s/%s. %v", apiVersion, kind, rNamespace, err),
+			})
+		}
+
+		for _, rName := range resources.Items {
+			if rNamespace == namespace {
+				log.V(4).Info("skip resource self-clone")
+				//		return nil, Skip, nil
+				response = append(response, GenerateResponse{
+					Data:   nil,
+					Action: Skip,
+					Error:  nil,
+				})
+			}
+
+			// check if the resource as reference in clone exists?
+			obj, err := client.GetResource(apiVersion, kind, rNamespace, rName.GetName())
+			if err != nil {
+				log.Error(err, "failed to get resoruce", apiVersion, "apiVersion", kind, "kind", rNamespace, "rNamespace", rName.GetName(), "name")
+				//				return nil, Skip, fmt.Errorf("source resource %s %s/%s/%s not found. %v", apiVersion, kind, rNamespace, rName.GetName(), err)
+				response = append(response, GenerateResponse{
+					Data:   nil,
+					Action: Skip,
+					Error:  fmt.Errorf("source resource %s %s/%s/%s not found. %v", apiVersion, kind, rNamespace, rName.GetName(), err),
+				})
+				return response
+			}
+
+			// remove ownerReferences when cloning resources to other namespace
+			if rNamespace != namespace && obj.GetOwnerReferences() != nil {
+				obj.SetOwnerReferences(nil)
+			}
+
+			// check if resource to be generated exists
+			newResource, err := client.GetResource(apiVersion, kind, namespace, rName.GetName())
+			if err == nil {
```

Comment:
Does `err==nil` mean the resource exists? It might be safer to check if `newResource` is empty as the error could be with other types. i.e., the API-server is unable to serve the request, etc.

Reply:
If there are other failure, in that case still the err will not be nil.  But adding the `newResource` nil check is good idea